File: pyg_1_3/pyg_parent/pyg_user_web/src/main/java/com/pyg/user/controller/UserController.java
package com.pyg.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.pojo.TbUser;
import com.pyg.user.service.UserService;
import com.pyg.util.PhoneFormatCheckUtils;
import entity.Result;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/user")
public class UserController {

	@Reference
	private UserService userService;
	@Autowired
	private RedisTemplate redisTemplate;

	/**
	 * 增加
	 * @param user
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbUser user, String checkCode){
		try {
			// 判断验证码是否一致
			String smsCode = (String) redisTemplate.boundHashOps("smsCode").get(user.getPhone());
			if (!checkCode.equals(smsCode)){
				return new Result(false, "注册码不正确");
			}
			userService.add(user);
			return new Result(true, "注册成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "注册失败");
		}
	}

	/**
	 * 发送验证码
	 * @param phone 手机号
	 * @return 操作结果
	 */
	@RequestMapping("/sendCode")
	public Result sendCode(String phone){
		// 校验手机号是否合法
		if (!PhoneFormatCheckUtils.isChinaPhoneLegal(phone)){
			return new Result(false, "手机号不合法");
		}
		try {
			userService.sendCode(phone);
			return new Result(true, "验证码已发送");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "验证码发送失败");
		}

	}
}

File: pyg_1_3/pyg_parent/pyg_user_web/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,userService){
	// 初始化entity对象
    $scope.entity = {};
	// 注册
	$scope.register = function () {
        // 用户名不能为空
        if ($scope.entity.username==null || $scope.entity.username==""){
            alert("请输入用户名");
            return;
        }
		// 密码不能为空
        if ($scope.entity.password==null || $scope.entity.password==""){
            alert("请输入密码");
            return;
        }
		// 校验密码是否一致
		if ($scope.entity.password!=$scope.password2){
            alert("密码不一致");
            return;
		}
        // 手机号不能为空
        if ($scope.entity.phone==null || $scope.entity.phone==""){
            alert("请输入手机号");
            return;
        }
		// 验证码不能为空
		if ($scope.checkCode==null || $scope.checkCode==""){
            alert("请输入验证码");
            return;
		}

		userService.add($scope.entity,$scope.checkCode).success(
			function (response) {
				if (response.success){
                    alert(response.message);
				} else {
                    alert(response.message);
				}
            }
		)
    };

    // 获取短信验证码
	$scope.sendCode = function () {
		// 检查手机号是否合法
		if ($scope.entity.phone==null || $scope.entity.phone==""){
            alert("请输入您的手机号");
            return;
		}
		userService.sendCode($scope.entity.phone).success(
			function (response) {
                if (response.success){
                    alert(response.message);
                } else {
                    alert(response.message);
                }
            }
		)
    };
});
